### PR TITLE
hopefully better ticket matcher

### PR DESCRIPTION
**Pull request by sevgibson**
[BUG][PLD-1234]

Was matching the entire thing (.\* matches BUG][PLD-1234) and also matching PLD-1234.

Restrict ticket id format so .\* will not grab everything...

### CODE REVIEW

File: scripts/pipeline_workflow.coffee [hunk 1]
```diff
@@ -164,7 +164,7 @@ module.exports = (robot) ->
       json = JSON.parse body
       console.log("json is ", json)
       title = json['title']
-      matches = title.match(/\[.*?\]/g)
+      matches = title.match(/\[[A-Z]*-[0-9]*?\]/g)
```
Comment (chadoh) [inline scripts/pipeline_workflow.coffee]: Should we do a white-list of our JIRA project prefixes and stop using the `[` symbols altogether? (People can keep putting them in if they prefer, but making queueing our regex matcher off of them seems needlessly brittle.)

``` coffeescript
# top-level constant
JiraProjects = "(PLD|INT|MBL|whatever)"

# later, in function
matches = title.matche(/#{JiraProjects}-[0-9]+/g)
```

Comment (sevgibson) [inline scripts/pipeline_workflow.coffee]: Well, we had .* -- which wasn't working at all for multiple tickets or [BUG][PLD-1234]

This is an improvement on that...

I'm not sure we want to design it so anytime anyone creates some new "project", we have to make an edit here for PRs to work...

Comment (chadoh) [inline scripts/pipeline_workflow.coffee]: Ok, well, I don't think you want to use `*` in your new pattern. That seems like bad news—0 or more! `+` will give us 1 or more, which is probably already too permissive, for the letter part (I think JIRA requires it to be at least two characters long).

``` coffeescript
/\[[A-Z]{2,}-[0-9]+?\]/g
```

**Comment by jeffoberlander** [comment]
@sevgibson Can you delete this old one and the branch if its no longer necessary?

**Pull request closed without merging**